internal/controllers/secrets.go

```diff
@@ -0,0 +1,134 @@
+// Copyright The Cryostat Authors
+//
+// The Universal Permissive License (UPL), Version 1.0
+//
+// Subject to the condition set forth below, permission is hereby granted to any
+// person obtaining a copy of this software, associated documentation and/or data
+// (collectively the "Software"), free of charge and under any and all copyright
+// rights in the Software, and any and all patent rights owned or freely
+// licensable by each licensor hereunder covering either (i) the unmodified
+// Software as contributed to or provided by such licensor, or (ii) the Larger
+// Works (as defined below), to deal in both
+//
+// (a) the Software, and
+// (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+// one is included with the Software (each a "Larger Work" to which the Software
+// is contributed by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition:
+// The above copyright notice and either this complete permission notice or at
+// a minimum a reference to the UPL must be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1beta1 "github.com/cryostatio/cryostat-operator/api/v1beta1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *CryostatReconciler) reconcileSecrets(ctx context.Context, cr *operatorv1beta1.Cryostat) error {
+	err := r.reconcileGrafanaSecret(ctx, cr)
```

Comment:
Should we generate Grafana Secret if deployment is minimal? Also, any clean up if deployment is switched to minimal from non-minimal?

Reply:
Ideally no, but there is no harm in doing so. I'll see if I can change it quickly.

Reply:
I've updated this PR to only create/update the secret for the non-minimal configuration, and delete the secret for the minimal configuration. This also removes the Secret link from the CR Status when deleting the secret.

I had to revise the password generation in the tests to accommodate this by adding a slice of generated passwords to the test input. Seems to work well.

Reply:
Great! Thanks for making edits... :D

Reply:
Right, that would make sense to do. I think that might be getting a bit beyond the scope of this PR since that's handled by the certificates. I filed #471 for this.

Reply:
Right make sense! Thanks :D